### ISSUE TITLE
Set config instance when calling attach

### ISSUE THREAD

**Issue by sk-** [JustinBeckwith/retry-axios#121]
The `raxConfig` requires an `instance` field, which if not set will default to the default axios instance, however that's not what one wants when attaching it to a non-default instance.

This is awkward an error prone, as one would need to create the object as follows:

```ts
const axiosInstance = axios.create({
  raxConfig: {
    retryDelay: 10000,
    backoffType: 'static',
  },
});
axiosInstance.defaults.raxConfig = {...axiosInstance.defaults.raxConfig, instance: axiosInstance};
```

Instead, I suggest to modify attach to do something like:

```ts
export function attach(instance?: AxiosInstance) {
  if (instance) {
    instance.defaults.raxConfig = {...instance.defaults.raxConfig, instance};
  }
  instance = instance || axios;
  return instance.interceptors.response.use(onFulfilled, onError);
}
```

**Comment by reessm01** [comment]
Yeah, I lost a bit of time figuring out that nuanced detail as I was limited to axios 0.19.2. How active is @JustinBeckwith? I wouldn't mind helping out as I think this library has a great deal of potential.

**Comment by JustinBeckwith** [comment]
👋  I am on GitHub most days for work, but do not actively do much on this library.  We ended up going a [different direction](https://github.com/googleapis/gaxios/) for our retries & HTTP, so anything I can give here is best effort.  I really love well thought out PRs though, and I'd be happy to add maintainers if that makes sense.

**Comment by reessm01** [comment]
I've got a feature branch ready whenever you get a chance if you want to add me as a maintainer. Beyond unit testing I'll try out the build in a project I am working on to see if its compatible with axios ~0.19.0.

**Comment by JustinBeckwith** [comment]
Closing as this is resolved by #312.
